Charon/filetypes/UltimakerFormatPackage.py

```diff
@@ -61,13 +61,13 @@ def openStream(self, stream: BufferedIOBase, mime: str = "application/x-ufp", mo
         self._readMetadata()  # Load the metadata, if any.
 
     def close(self) -> None:
-        if not self.stream:
+        if not self.stream or not self.zipfile:
             raise ValueError("This file is already closed.")
         self.flush()
         self.zipfile.close()
 
     def flush(self) -> None:
-        if not self.stream:
+        if not self.stream or not self.zipfile:
```

Comment:
Nitpicking; is it intentional that you use not self.zipfile and self.zipfile is None? Is this intentional? If not, i think it's better to be consistent (and use self.zipfile is None)

Reply:
Actually, these should be just asserts, if stream is not set, then zipfile is also not set. If stream is available and zipfile is not, that indicates an code malfunction.